delta/kernel/src/main/scala/ch/epfl/bluebrain/nexus/delta/kernel/utils/ClasspathResourceLoader.scala

```diff
@@ -106,8 +117,17 @@ class ClasspathResourceLoader private (classLoader: ClassLoader) {
       jsonObj <- IO.fromOption(json.asObject)(InvalidJsonObject(resourcePath))
     } yield jsonObj
 
-  private def resourceAsTextFrom(resourcePath: String): IO[String] =
-    streamOf(resourcePath).map(is => Source.fromInputStream(is)(Codec.UTF8).mkString)
+  private def resourceAsTextFrom(resourcePath: String): IO[String] = {
+    fs2.io
+      .readClassLoaderResource[IO](resourcePath, classLoader = classLoader)
+      .through(text.utf8.decode)
+      .compile
+      .string
+      .adaptError {
+        case e: IOException if Option(e.getMessage).exists(_.endsWith("not found")) =>
+          ResourcePathNotFound(resourcePath)
```

Comment:
Not sure the `ResourcePathNotFound` brings much here

Reply:
The reason is only that it's a `Rejection`, and it can happen in production code so it's probably nicer. There was a test asserting that this was the error